src/Examples/docs_examples.jl

```diff
@@ -41,9 +41,9 @@ function draw_zoomlenses(filenames::Vector{<:Union{Nothing,AbstractString}} = re
     raygenerator = Sources.Source(; transform, origins, directions)
 
     aspherics = [
-        [0, 0, 0, 1.0386E-04, 0, 1.4209E-07, 0, -8.8495E-09, 0, 1.2477E-10, 0, -1.0367E-12, 0, 3.6556E-15],
-        [0, 0, 0, 4.2721E-05, 0, 1.2484E-07, 0, 9.7079E-09, 0, -1.8444E-10, 0, 1.8644E-12, 0, -7.7975E-15],
-        [0, 0, 0, 1.1339E-04, 0, 4.8165E-07, 0, 1.8778E-08, 0, -5.7571E-10, 0, 8.9994E-12, 0, -4.6768E-14],
+        [4 => 1.0386E-04, 6 => 1.4209E-07, 8 => -8.8495E-09, 10 => 1.2477E-10, 12 => -1.0367E-12, 14 => 3.6556E-15],
+        [4 => 4.2721E-05, 6 => 1.2484E-07, 8 => 9.7079E-09, 10 => -1.8444E-10, 12 => 1.8644E-12, 14 => -7.7975E-15],
+        [4 => 1.1339E-04, 6 => 4.8165E-07, 8 => 1.8778E-08, 10 => -5.7571E-10, 12 => 8.9994E-12, 14 => -4.6768E-14],
```

Comment:
Not quite a dictionary, but I think Pairs here are more expressive than tuples regarding our intended usage. This will easily convert to a Dict if needed later.

Reply:
Yeah I think there is a better way to do this still eg where surface type is the actual type name allowing you to get the constructor for the surface dynamically and then you could pass the parameter dict as keyword args. This is a pretty fundamental change and might mean modifying the surface constructors too (ughh) but would mean that new surface types would be supported directly and there wouldn’t be any need for hard coding stuff when setting up axisymmetric systems. 

Anyway this seems reasonable for now, but I think there is a better solution there that might be easiest to achieve by starting almost from scratch on the design of AxisymmetricOpticalSystem

Reply:
I see what you mean - got the wrong idea earlier!

It'll definitely require a lot of changes, and it'll get mixed up with some other discussions we're having now about prescription DataFrames for non-axisymmetric systems... (although this also means it's a good time for a rewrite). Let's go with this for now - in the short term, all I want is to add support for Zernike surfaces. I'll factor this into our discussions and work towards something smoother in the future.